src/main/java/com/minecolonies/coremod/blocks/BlockHutTownHall.java

```diff
@@ -44,8 +45,12 @@ public void onBlockPlacedBy(
         if(placer.getActiveHand().equals(EnumHand.MAIN_HAND))
         {
             final Colony colony = ColonyManager.getClosestColony(worldIn, pos);
-            if (colony == null && placer instanceof EntityPlayer)
+
+            if ((colony == null
+                    || BlockPosUtil.getDistance2D(colony.getCenter(), pos) >= Configurations.workingRangeTownHall * 2 + Configurations.townHallPadding)
```

Comment:
He only reaches this if the placement is valid. But we can't be sure whether he is placing it again or the first time. (Inside the correct range)